client_it_test.go

```diff
@@ -573,3 +574,68 @@ func TestClientVersion(t *testing.T) {
 	// adding this test here, so there's no "unused lint warning.
 	assert.Equal(t, "1.1.0", hz.ClientVersion)
 }
+
+func TestInvocationTimeout(t *testing.T) {
+	tc := it.StartNewClusterWithOptions("invocation-timeout", 41701, 1)
+	defer tc.Shutdown()
+	config := tc.DefaultConfig()
+	if it.TraceLoggingEnabled() {
+		config.Logger.Level = logger.TraceLevel
+	}
+	if it.NonSmartEnabled() {
+		config.Cluster.Unisocket = true
+	}
+	const timeout = 3 * time.Second
+	config.Cluster.InvocationTimeout = types.Duration(timeout)
+	client, err := hz.StartNewClientWithConfig(context.Background(), config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := context.Background()
+	myMap, err := client.GetMap(ctx, "my-map")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tc.Shutdown()
+	tic := time.Now()
+	_, err = myMap.Get(ctx, "k1")
+	took := time.Since(tic)
+	if took > timeout+1*time.Second {
```

Comment:
This test will fail when there is a load on the system. I think it is ok to see that it returned an error eventually. Or make the timeout 2 minutes again if you really want to assert time.

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/655/commits/24c7e6c984d78284d27d169348faa7438dfb1dee